[FL-870] Auto-generated firmware documentation

drunkbatya opened the pull request:
# What's new

- Add generating documentation CI/CD on push to dev branch
- Add uploading documentation CI/CD to Github Pages

# Verification 

- Push something to dev branch, updated documentation will be available on 'https://flipperdevices.github.io/flipperzero-firmware/'

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

Commit:
Add doxygen and doxygen-awesome css, cleanup docs files

Commit:
Ignore more libraries and remove leftover local variables

Commit:
Create an actual intro page

Commit:
Merge branch 'dev' into astra/docs-autogen

Commit:
.md files linting

drunkbatya committed:
Add doxygen action

drunkbatya committed:
Fix Doxygen path

drunkbatya committed:
Fix doxyfile path

drunkbatya committed:
Try to upload

drunkbatya committed:
Change docs branch

drunkbatya committed:
Add submudules checkout

drunkbatya committed:
Disable doxygen on PR

Commit:
Mention the firmware docs in the readme

Commit:
More dev docs mentions in the readme

skotopes committed:
Merge branch 'dev' into astra/870-docs-autogen

skotopes commented on .github/workflows/docs.yml:
```diff
@@ -0,0 +1,55 @@
+jobs:
+  doxygen:
+    if: ${{ !github.event.pull_request.head.repo.fork }}
+    runs-on: [self-hosted, FlipperZeroShell]
```
I don't see reasons why we should run it on our runners.

skotopes commented on .github/workflows/docs.yml:
```diff
@@ -0,0 +1,55 @@
+on:
+  push:
+    branches:
+      - dev
```
We do need tags too:
- Production: all tags separately
- Dev: overwriting

drunkbatya committed:
Fix runner group, add tags

drunkbatya committed:
Test dev in PR

drunkbatya committed:
Disable running on PR

giacomoferretti commented on documentation/Doxyfile.cfg:
```diff
@@ -2021,7 +2035,7 @@ RTF_EXTENSIONS_FILE    =
 GENERATE_MAN           = NO
 
-# The MAN_OUTPUT tag is used to specify where the man pages will be put. If a
+# The MAN_OUTPUT tag is used to specify where the man И will be put. If a
```
Out of curiosity, why this change? I know it's just a comment and that it doesn't impact the docs generation.

skotopes replied:
looks like some replace artifact

Astrrra replied:
Oops, I think I just accidentally typed that, fixed now

Astrrra committed:
Fix a typo in the doxyfile

drunkbatya committed:
Try upload to S3

drunkbatya committed:
Merge branch 'dev' into astra/870-docs-autogen

drunkbatya committed:
Fix local path

drunkbatya committed:
Fix S3 ACL

drunkbatya committed:
Add delete flag, unifying dev and tags

drunkbatya committed:
Remove delete flag

drunkbatya committed:
Remove PR trigger

Astrrra committed:
Merge branch 'dev' into astra/870-docs-autogen

Astrrra committed:
Update submodules

Astrrra committed:
Initial sphinx test

Astrrra committed:
Gitignore even more + don't generate latex

Astrrra committed:
Sphinx-multiversion made me do this

I had to commit the whole doxygen output because sphinx-multiversion doesn't seem to work otherwise -_-

Astrrra committed:
I now have to commit each and every change to test

Astrrra committed:
One more testing commit

Astrrra committed:
More testing commits

Astrrra committed:
Try generating directly from the doxygen folder

Astrrra committed:
Fixing my testing sins

Astrrra committed:
Restructure gitignores

Astrrra committed:
More ignoring and more configs

Astrrra committed:
Unfortunately the .rst files have to be committed

Astrrra committed:
I am deeply upset by the fact that I have to add all this as well... Sorry...

Astrrra committed:
Read markdown directly from sphinx

Astrrra committed:
Move new docs and markdown linting

Astrrra committed:
Finding index

Astrrra committed:
Fix typo

Astrrra committed:
Index needs to be in rst -_-

Astrrra committed:
Header broke and the contents are excessive

Astrrra committed:
The list of contents is needed for the sidebar

Astrrra committed:
Experimenting with directives

Astrrra committed:
More experimenting

Astrrra committed:
Even more experimenting

Astrrra committed:
Contents can't have a caption

Astrrra committed:
Remove content from the contents directive

Astrrra committed:
The contents directive is just broken

Astrrra committed:
Contents have to be typed in manually by hand

Astrrra committed:
Create the contents

Astrrra committed:
Fix contents

Astrrra committed:
More fixing of contents

Astrrra committed:
And more fixes for contents

Astrrra committed:
Move the templates back

Astrrra committed:
Shallower depth of the TOC

Astrrra committed:
Even shallower TOC

Astrrra committed:
Update doxygen files

Astrrra committed:
Delete conf.py

Astrrra committed:
Update doxygen files

Astrrra committed:
Update doxysphinx files

Astrrra committed:
Add the description for auto-generated pages

Astrrra committed:
Update doxysphinx files

Astrrra committed:
Update index.rst

Astrrra committed:
Update index.rst

Astrrra committed:
Update index.rst

Astrrra committed:
Update index.rst

Astrrra committed:
Move most docs to a submodule

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Removed submodule documentation/flipperzero-documentation

Astrrra committed:
Move documentation to a submodule (for real this time)

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Move .gitignore to submodule, move submodule to the root dir

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Update docs

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Merge branch 'astra/docs-branches-test' into astra/870-docs-autogen

Astrrra committed:
Merge remote-tracking branch 'origin/dev' into astra/870-docs-autogen

Astrrra committed:
Update docs contents

drunkbatya committed:
Merge branch 'dev' into astra/870-docs-autogen

drunkbatya committed:
Trying to install requirements

drunkbatya committed:
Temporary enable PR action

drunkbatya committed:
Trying to build the docs

Astrrra committed:
Update flipperzero-documentation

drunkbatya committed:
Add uploading to GithubPages

drunkbatya committed:
Update html path

drunkbatya committed:
Debug printf

drunkbatya committed:
List level up

drunkbatya committed:
Add debug to sphinx-multiversion

drunkbatya committed:
Trying to debug

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Update flipperzero-documentation

Astrrra committed:
Update flipperzero-documentation

drunkbatya committed:
Unsing patched sphinx-multiversion

drunkbatya committed:
Change requrements install priority

drunkbatya committed:
Deb it ug

Astrrra committed:
Update flipperzero-documentation

drunkbatya committed:
Uploading docs from parent directory

drunkbatya committed:
Uploading docs from branch directory

drunkbatya committed:
Update flipperzero-documentation, add index.html

drunkbatya committed:
Disable sphinx-multiversion debug

drunkbatya committed:
Disable generating on PR's

drunkbatya committed:
Merge branch 'dev' into astra/870-docs-autogen

Astrrra committed:
Update protobuf to dev

Astrrra committed:
Build docs only for dev and releases

Astrrra committed:
Move docs out of the submodule

Astrrra committed:
Update docs submodule

Astrrra committed:
Re-enable generating on PRs

Astrrra committed:
Fix docs paths

Astrrra committed:
Fix docs paths again

Astrrra committed:
Even more docs paths fixing

Astrrra commented:
Superseded by #2944
